File: triatlon-cs/Model/Person/Referee.cs


// Referee

public class Referee : Person
{

    // Properties
    
    private RaceType _raceType;
    private string _email;
    private string _password;

    public RaceType RaceType
    {
        get => _raceType;
    }

    public string Email
    {
        get => _email;
    }

    public string Password
    {
        get => _password;
    }

    // Lifecycle

    public Referee(string firstName, string lastName, RaceType raceType, string email, string password) : base(
        firstName, lastName)
    {
        _raceType = raceType;
        _email = email;
        _password = password;
    }

    // Override Methods

    public override string ToString()
    {
        return "Referee | " + base.ToString();
    }

}

File: triatlon-java/model/src/main/java/triatlon/model/person/Referee.java
package triatlon.model.person;

import triatlon.model.activity.RaceType;

import java.io.Serializable;

// Referee

public class Referee extends Person implements Serializable {

    // Private Properties

    private final RaceType raceType;
    private final String email;
    private final String password;

    // Lifecycle

    public Referee(String firstName, String lastName, RaceType raceType, String email, String password) {
        super(firstName, lastName);
        this.raceType = raceType;
        this.email = email;
        this.password = password;
    }

    // Getters

    public RaceType getRaceType() {
        return raceType;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    // Override Methods

    @Override
    public String toString() {
        return "Referee | " + super.toString();
    }

}
